File: Visual Basic .NET/Project Euler #087.vb
Module Module1    

    Function isPrime(ByVal n As Integer) As Integer 'Función para determinar si un número es primo.
        If n = 1 Then
            Return False
        ElseIf n < 4 Then
            Return True
        ElseIf n Mod 2 = 0 Then
            Return False
        ElseIf n < 9 Then
            Return True
        ElseIf n Mod 3 = 0 Then
            Return False
        Else
            Dim r As Integer = Math.Floor(Math.Sqrt(n))
            Dim f As Integer = 5
            Do While f <= r
                If n Mod f = 0 Then
                    Return False
                ElseIf n Mod (f + 2) = 0 Then
                    Return False
                Else
                    f += 6
                End If
            Loop
            Return True
        End If
    End Function

    Sub Main()

        Dim n As Integer = 1
        Dim primo1 As Integer = 2
        Dim primo2 As Long = 2
        Dim primo3 As Long = 2
        Dim primomin As Long = 2
        Dim primomin2 As Long = 2
        Dim suma As Long = 0
        Dim limite As Integer = 50000000
        Dim respuesta As Integer = 0

        'Busca cuántos números menores a cincuenta millones pueden expresarse como la suma de una cuadrado, un cubo y un fourth power primos

        Do Until primo2 ^ 3 > limite And primo3 ^ 4 > limite

            'Cuando se encuentre un primo, se suma a la respuesta

            If isPrime(n) Then
                primo1 = n
                suma = primo1 ^ 2 + primo2 ^ 3 + primo3 ^ 4
                If suma <= limite Then
                    respuesta += 1
                End If
            End If

            n += 1

            'Si el cubo es mayor al límite pero el fourt power entonces se repite el proceso
            If primo2 ^ 3 > limite And limite > primo3 ^ 4 Then

                Console.Write(n)
                Console.Write("   ")
                Console.Write(respuesta)
                Console.Write("   ")
                Console.Write(primo1 ^ 2)
                Console.Write("   ")
                Console.Write(primo2 ^ 3)
                Console.Write("   ")
                Console.Write(primo3 ^ 4)
                Console.Write("   ")
                Console.Write(suma)
                Console.Write("   ")
                Console.Write(primomin)
                Console.Write("   ")
                Console.WriteLine(primomin2)
                'Console.ReadLine()

                n = 2
                primo1 = 2
                primo2 = 2 'La base del cubo vuelve al siguiente primo
                primo3 = primomin2 'La base del fourth power vuelve al siguiente primo
                primomin = 2 'La base del fourth power será menor que la del cubo

                Do Until primo3 > primomin2 And isPrime(primo3)
                    primo3 = primo3 + 1
                Loop
                primomin2 = primo3 'Se consigue el próximo primo para repetir el proceso

            ElseIf suma > limite And limite > primo2 ^ 3 Then  'Cuando la suma es mayor que el límite pero el cubo es menor al límite
                primo1 = 2 'La base del cuadrado regresa a 2, el menor primo
                n = 2 'Se vuelven a buscar primos desde 3
                primo2 = primomin 'La base del cubo vuelve al siguiente primo
                Do Until primo2 > primomin And isPrime(primo2)
                    primo2 = primo2 + 1
                Loop
                primomin = primo2 'Se escoge una nueva base prima

            ElseIf suma > limite And suma < primo2 ^ 3 Then
                primo1 = 2
                primo2 = 2
            End If
        Loop

        Console.Write("La respuesta es ")
        Console.Write(respuesta)
        Console.ReadLine()

    End Sub
End Module

File: C/Project Euler #087.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int isPrime(int n);

/* Solución para el problema #87 de Project Euler
 * "¿Cuántos números menores de cincuenta millones pueden expresarse como la suma de un cuadrado primo, un cubo primo y una cuarta potencia prima?"
 * https://projecteuler.net/problem=87            */

int main()
{
    int x, y, z, n, respuesta = 0;

    for (n=1; n<=50000000 ; n++){ //50000000
        for (x=2; (x*x*x*x) <= n ; x++){
            if (isPrime(x) == 1){
                for (y=2; (y*y*y) <= n ; y++){
                    if (isPrime(y) == 1){
                        for (z=2; (z*z) <= n ; z++){
                            if (isPrime(z) == 1){
                                if (((x*x*x*x) + (y*y*y) + (z*z)) == n){
                                    respuesta++;
                                    printf("n:%d, Respuesta:%d (x:%d y:%d z:%d)\n", n, respuesta, x, y, z);
                                }
                            }
                        }
                    }
                }
            }
        }
        if (n % 1000000 == 0) printf("n:%d, Respuesta:%d\n", n, respuesta);
    }
    printf("\nLa respuesta es:%d\n", respuesta);
    printf("Fin del programa\n");
    return 0;
}

int isPrime(int n){ //Función para determinar si un numero es primo.
    if (n == 1) return 1; ////*Cambiar a 0 o mantener a conveniencia del programa
    else if (n < 4) return 1; //Si son menores de 4, son primos
    else if (n % 2 == 0) return 0; //Si son pares, no son primos
    else if (n < 9) return 1; //Si son menores de 9 e impares, son primos
    else if (n % 3 == 0) return 0; //Si son múltiplos de 3, no son primos
    else{ //Criva de Erastóstenes
        int root = floor(sqrt(n));
        int i = 5;
        while (i <= root){
            if (n % i == 0) return 0;
            else if (n % (i + 2) == 0) return 0;
            else i += 6; //11, 17
        }
        return 1;
    }
}
